src/devices/Common/Iot/Device/Multiplexing/IOutputSegment.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Threading;
+
+namespace Iot.Device.Multiplexing
+{
+    /// <summary>
+    /// Interface that abstracts multiplexing over a segment of outputs.
+    /// </summary>
+    public interface IOutputSegment : IDisposable
+    {
+        /// <summary>
+        /// Length of segment (number of outputs)
+        /// </summary>
+        int Length { get; }
+
+        /// <summary>
+        /// Segment values.
+        /// </summary>
+        PinValue this[int index] { get; }
+
+        /// <summary>
+        /// Writes a PinValue to a virtual output.
+        /// Does not display output.
+        /// </summary>
+        void Write(int output, PinValue value);
+
+        /// <summary>
+        /// Writes discrete underlying bits to a virtual output.
+        /// Written one byte at a time, left to right. Least significant bit will written to index 0.
+        /// Does not display output.
+        /// </summary>
+        void Write(int value);
+
+        /// <summary>
+        /// Turns off all outputs.
+        /// Performs a latch.
+        /// </summary>
+        void Clear();
+
+        /// <summary>
+        /// Displays segment until token receives a cancellation signal, possibly due to a specificated duration.
+        /// Publishes (latches) values.
+        /// </summary>
+        void Display(CancellationToken token);
```

Comment:
Maybe this should be Async or have an Async overload?

Reply:
Adding overload.